lib/node-includes-overrides.js

```diff
@@ -0,0 +1,34 @@
+/**
+ * Uses eslint (private) api to see if eslint config includes a `overrides` flag.
+ *
+ * inputNode type is a bit tricky here. We are expecting three different cases:
+ * if it's string, we will use it as a directory and let eslint do the work.
+ *
+ * if it's a source node, use the `sourceDirectory` property.
+ *
+ * if it's a transform node, recursively call same function with all of its
+ * inputNodes. if any of the input directories contain `overrides` eslint
+ * config, return true
+ *
+ * While it's not ideal we iterate through all the `inputNodes` of a transform node,
+ * the input size should be very small.
+ */
+const nodeIncludesOverrides = (inputNode, eslintCli) => {
+  if (typeof inputNode === 'string') {
+    return eslintCli.config
+      .getConfigHierarchy(inputNode)
+      .some(node => node.overrides);
+  }
+
+  const { nodeType, sourceDirectory } = inputNode.__broccoliGetInfo__();
+  if (nodeType === 'transform') {
+    const nodesIncludesOverrides = inputNode._inputNodes.map(node =>
+      nodeIncludesOverrides(node, eslintCli)
+    );
+    return nodesIncludesOverrides.some(Boolean);
+  }
+
+  return nodeIncludesOverrides(sourceDirectory, eslintCli);
```

Comment:
@rwjblue 

This is a bit questionable, mostly due to my lack of knowledge in Broccoli.

I feel it's safe to do 

```
if (nodeType === 'transform') {
...
} else if (nodeType === 'source') {
...
}

return false
```

Let me know